### ISSUE TITLE
`~` shorthand for home directory does not work

### ISSUE THREAD

**Issue by Ascyt** [xremap/xremap#534]
I'm using Arch linux with wayland and hyprland. All the code snippets here are under `keymap`, since it's not possible to remap keys to CapsLock otherwise.

It works if I have:

```yml
      CapsLock-n: 
        launch: ["bash", "-c", "/home/ascyt/.config/xremap/script/script.sh"]
```

However, if I instead try to do the following:

```yml
      CapsLock-n: 
        launch: ["bash", "-c", "~/.config/xremap/script/script.sh"]
````

It does not work, and nothing gets executed. `~` is in linux a shorthand for the home directory of the current user, in my case generally meaning the exact same as `/home/ascyt`. Using `$HOME` in place of `~` does not work either.

**Comment by wistoft** [comment]
When I try it out `~` works for me. And xremap has not much to do with it. Because it's an argument to bash, which should interpret `~`.
The only thing I can come up with is xremap is running with a different user than `ascyt`, so bash looks in a different home folder. Can you check that?

**Comment by Ascyt** [comment]
It is the correct user, and it's the same whether I run it in vscode or in the terminal. Processes down the line (eg. a Python script getting run by a bash script that uses `~`) also don't work. 

I don't know. I guess it's not a big deal anyways, just a minor annoyance. I guess might depend on and/or be an issue of the distro, windowing system etc.

**Comment by wistoft** [comment]
Okay. I'm not sure what it can be either. A thing you could try is to print the value of `~` to a file. It might give you some clue:
```
launch: ["bash", "-c", "echo ~ > /home/ascyt/tilde.txt"]
```

**Comment by Ascyt** [comment]
It prints `/root` to the file. It might have to do with the fact I'm running xremap with sudo?

**Comment by k0kubun** [comment]
> It might have to do with the fact I'm running xremap with sudo?

Yes it does. sudo would change the home directory under the process, but I guess you could change `HOME` environment variable, e.g. with `HOME=/home/ascyt sudo -E xremap ...`
